### ISSUE TITLE
F841 - False positive

### ISSUE THREAD

**Issue by Mister7F** [charliermarsh/ruff#2481]
Found this issue while playing with ruff, the variable `exponential` is used

```py
def f():
    exponential, base_multiplier = 1, 2
    hash_map = {
        (exponential := (exponential * base_multiplier) % 3): i + 1 for i in range(2)
    }
    return hash_map

# ruff --select=F test.py
# test.py:2:5: F841 Local variable `exponential` is assigned to but never used
```

**Comment by charliermarsh** [comment]
Interesting, thanks!

**Comment by charliermarsh** [comment]
Flake8 misses this too.
